### PR TITLE
x11-terms/cool-retro-term: Bump cool-retro-term to 1.2.0-r1 and added 9999 ebuild

### PR DESCRIPTION
**Pull request by thamognya**
Changes:
- added 1.2.0 version ebuild
- added 9999 version ebuild
- Added Manifest updates
- EAPI to 8
- Requesting to be a proxy-maint or proxy-maintainer or maintainer for this package `x11-terms/cool-retro-term`

Signed-off-by: Thamognya Kodi <contact@thamognya.com>

### CODE REVIEW

File: x11-terms/cool-retro-term/cool-retro-term-1.2.0-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	popd || die
+}
+
+src_configure() {
+	eqmake5 PREFIX="${EPREFIX}/usr"
+}
+
+src_install() {
+	emake
+	mv "${WORKDIR}/${P}/${PN}" "/usr/bin/${PN}"
```
Comment (juippis) [inline x11-terms/cool-retro-term/cool-retro-term-1.2.0-r2.ebuild:49]: You'll most likely want to replace this with something like:
```
	emake INSTALL_ROOT="${D}" install
	einstalldocs
```
`"/usr/bin/${PN}"` causes a sandbox error, you'll have to prepend it with `"${ED}/usr/bin/${PN}"`

It's good to take a look at how other packages using this eclass handle building software.
https://qa-reports.gentoo.org/output/eclass-usage/qmake-utils.txt

File: x11-terms/cool-retro-term/cool-retro-term-1.2.0-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+pkg_postinst() {
+	xdg_icon_cache_update
+	xdg_desktop_database_update
+}
+
+pkg_postrm() {
+	rm -rf "/usr/bin/${PN}" || die
```
Comment (juippis) [inline x11-terms/cool-retro-term/cool-retro-term-1.2.0-r2.ebuild:58]: This looks rather scary too.

File: x11-terms/cool-retro-term/cool-retro-term-1.2.0-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+DESCRIPTION="terminal emulator which mimics the look and feel of the old cathode tube screens"
+HOMEPAGE="https://github.com/Swordfish90/cool-retro-term"
+
+SRC_URI="https://github.com/Swordfish90/cool-retro-term/archive/${PV}.tar.gz -> ${P}.tar.gz
+		https://github.com/Swordfish90/qmltermwidget/archive/${QTW_PV}.tar.gz -> ${QTW_P}.tar.gz"
+
+LICENSE="GPL-2 GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc64 ~x86"
+IUSE=""
```
Comment (juippis) [inline x11-terms/cool-retro-term/cool-retro-term-1.2.0-r2.ebuild:21]: This empty variable can be removed.

File: x11-terms/cool-retro-term/metadata.xml [hunk 4]
```diff
@@ -1,7 +1,14 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
 <pkgmetadata>
-	<!-- maintainer-needed -->
+	<maintainer type="person">
+		<email>contact@thamognya.com</email>
+		<name>Thamognya Kodi</name>
+	</maintainer>
```
Comment (juippis) [inline x11-terms/cool-retro-term/metadata.xml:7]: You know you don't **have** to become a maintainer to contribute for a package? Happy to merge drive-bys!

**Review by juippis — COMMENTED** [review]
Also please fix these issues in your git branch, squash the fixes to your original commit and **force push** the branch. No need to keep closing PRs and opening new ones, we lose this review history.

**Comment by ThamognyaKodi** [comment]
Ah I see the errors now, I have tried to follow the suggestions made. Thank you for the feedback as I am starting out with writing gentoo ebuilds.

**Comment by thesamesam** [comment]
Looking good! Could you squash the fixup commits, then force push?

File: x11-terms/cool-retro-term/cool-retro-term-9999.ebuild [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+	default
+}
+
+src_configure() {
+	eqmake5 PREFIX="${EPREFIX}/usr"
+}
+
+src_install() {
+	emake
+	mv "${WORKDIR}/${P}/${PN}" "/usr/bin/${PN}"
```
Comment (juippis) [inline x11-terms/cool-retro-term/cool-retro-term-9999.ebuild:38]: (FYI these issues are present in the -9999 still)
Do we want a -9999 ebuild for this?

**Review by juippis — COMMENTED** [review]
I don't know if it's me or this package, but I can't launch it.
```
# cool-retro-term 
QQmlApplicationEngine failed to load component
qrc:/main.qml:143:5: Type TerminalContainer unavailable
qrc:/TerminalContainer.qml:47:5: Type PreprocessedTerminal unavailable
qrc:/PreprocessedTerminal.qml:123:9: Cannot assign to non-existent property "blinkingCursor"
Cannot load QML interface
```
And this seems to be a rather common unsolved error, 
https://github.com/Swordfish90/cool-retro-term/issues/592
https://github.com/Swordfish90/cool-retro-term/issues/700
https://github.com/Swordfish90/cool-retro-term/issues/682
https://github.com/Swordfish90/cool-retro-term/issues/592
https://github.com/Swordfish90/cool-retro-term/issues/620
https://github.com/NixOS/nixpkgs/issues/12027

so I really don't know what to do here. All I can say is that 1.1.1-r2 works for me... so it's looking bad. Also you shouldn't revbump 1.2.0 to -r2 in this PR since it hasn't been introduced to the tree. 1.2.0 is fine.

**Comment by ThamognyaKodi** [comment]
I have been looking into the same problem on my side to. The issue is something I can not really find, but my original and 9999 ebuild I submitted with the non-devmanual / risky way of code fixed it, but as you said in your previous comments, they look a bit risky and even I as a end user would probably not use it. I am thinking of making a patch or such, because the issue lies in the Makefile (or there might be a extra step mentioned in one of the issues) of this version (at least from my research). I think I will try to fix on that or make this pr a draft and submit a working version soon.

**Comment by juippis** [comment]
The problem with the previous ebuild was that there's no way you could install it using portage due to sandbox protection. 

But if you've detected the root cause for this issue and can make a patch, that sounds very promising! It could also help upstream.
